src/renderer.test.ts

```diff
@@ -266,137 +274,152 @@ describe('Renders different types of elements', () => {
         const nodes = render(guideElement())
         const guide = nodes.flat()[0];
         expect(getHtml(guide)).toContain('<main>Guide content</main>');
         testHandlers(guide);
-    })
+   })
 
     test('ElementKind.QandaAtom', () => {
         const nodes = render(qandaElement())
         const qanda = nodes.flat()[0];
         expect(getHtml(qanda)).toContain('<main>QandA content</main>');
         testHandlers(qanda);
-    })
+   })
 
     test('ElementKind.ProfileAtom', () => {
         const nodes = render(profileElement())
         const profile = nodes.flat()[0];
         expect(getHtml(profile)).toContain('<main>Profile content</main>');
         testHandlers(profile);
-    })
+   })
 
     test('ElementKind.TimelineAtom', () => {
         const nodes = render(timelineElement())
         const timeline = nodes.flat()[0];
         expect(getHtml(timeline)).toContain('<p>Swedish prosecutors announce they are <a href="https://www.theguardian.com/media/2019/may/13/sweden-reopens-case-against-julian-assange">reopening an investigation into a rape allegation</a> against Julian Assange.</p><p><br></p>');
         testHandlers(timeline);
-    })
+   })
 
     test('ElementKind.ChartAtom', () => {
         const nodes = render(chartElement())
         const chart = nodes.flat()[0];
-        expect(getHtml(chart)).toContain('srcDoc=\"&lt;main&gt;Chart content&lt;/main&gt;\"');
-    })
+        expect(getHtml(chart)).toContain('srcDoc="&lt;main&gt;Chart content&lt;/main&gt;"');
+   })
+
+
 
     function testHandlers(node: ReactNode): void {
         if (isValidElement(node)) {
-            const component = mount(node);
-            const likeButton = component.find('[data-testid="like"]');
-            const dislikeButton = component.find('[data-testid="dislike"]');
-            const feedback = component.find('[data-testid="feedback"]');
-            expect(feedback.html()).toContain('hidden=""');
-            dislikeButton.simulate('click');
-            likeButton.simulate('click');
-            expect(feedback.html()).not.toContain('hidden=""');
-        }
-    }
+            const container = document.createElement("div");
+            document.body.appendChild(container);
+            act(() => {
+                renderDom(node, container);
+            });
+
+            const likeButton = container.querySelector('[data-testid="like"]');
+            const dislikeButton = container.querySelector('[data-testid="dislike"]');
+            const feedback = container.querySelector('[data-testid="feedback"]');
+            expect(feedback?.getAttribute('hidden')).toBe("")
+
+            act(() => {
+                dislikeButton!.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+            });
+            act(() => {
+                likeButton!.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+            });
+
+            expect(feedback!.textContent).not.toBe('hidden=""');
```

Comment:
I'm not sure `hidden` will ever appear in [`textContent`](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent) here because it's an attribute. You may want something like this instead?

```suggestion
            expect(feedback?.getAttribute('hidden')).toBeNull();
```